srml/executive/src/lib.rs

```diff
@@ -213,6 +214,50 @@ impl<
 		header.state_root().check_equal(&storage_root);
 		assert!(header.state_root() == &storage_root, "Storage root must match that calculated.");
 	}
+
+	/// Check a given transaction for validity. This doesn't execute any
+	/// side-effects; it merely checks whether the transaction would panic if it were included or not.
+	/// 
+	/// Changes made to the storage should be discarded.
+	pub fn validate_transaction(uxt: Block::Extrinsic) -> TransactionValidity {
+		let encoded_len = uxt.encode().len();
+		
+		let xt = match uxt.check_with(Lookup::lookup) {
+			// Checks out. Carry on.
+			Ok(xt) => xt,
+			// An unknown account index implies that the transaction may yet become valid.
+			Err("invalid account index") => return TransactionValidity::Unknown,
+			// Technically a bad signature could also imply an out-of-date account index, but
+			// that's more of an edge case.
+			Err(_) => return TransactionValidity::Invalid,
+		};
+
+		if let Some(sender) = xt.sender() {
+			// pay any fees.
+			if Payment::make_payment(sender, encoded_len).is_err() {
+				return TransactionValidity::Invalid
+			}
+
+			// check index
+			let mut expected_index = <system::Module<System>>::account_nonce(sender);
+			if xt.index() < &expected_index {
+				return TransactionValidity::Invalid
+			}
+			if *xt.index() > expected_index + As::sa(256) {
```

Comment:
So this would be specific to runtime, right? The pool should not make any assumptions like this?

Reply:
right - this is just a sanity check. the pool should be ignorant of such things and rely on what the runtime is telling it.